gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -465,15 +465,18 @@ public void write(String uploadId, byte[] toWrite, int toWriteOffset, long destO
       GenericUrl url = new GenericUrl(uploadId);
       HttpRequest httpRequest = storage.getRequestFactory().buildPutRequest(url,
           new ByteArrayContent(null, toWrite, toWriteOffset, length));
-      long limit = destOffset + length;
-      StringBuilder range = new StringBuilder("bytes ");
-      range.append(destOffset).append('-').append(limit - 1).append('/');
-      if (last) {
-        range.append(limit);
-      } else {
-        range.append('*');
+      // 0-length requests are valid, don't need content-range.
```

Comment:
Ah, I wish that in your case of `Files.createFile` you could have used the other create with zero byte array option (which would save an RPC).

Also, are you sure that the file was actually written to GCS. If so, does it mean that if you omit the content-range it is same as explicit finalize (suffixed with "/value" as oppose to "/*")?
In that case we only want to do it when `last == true`.  /cc @mziccard